### PR TITLE
Add "vision" to Python mode

### LINKED ISSUE #795: Open Files
### Describe the bug

I tried to open a screenshot with this command and Open Interpreter Crashed:

`> open /Users/maxpetrusenko/Desktop/photo_2023-11-23_17-58-50.jpg please

        Python Version: 3.10.12
        Pip Version: 23.2.1
        Open-interpreter Version: cmd:A, pkg: 0.1.15
        OS Version and Architecture: macOS-14.1-arm64-arm-64bit
        CPU Info: arm
        RAM Info: 16.00 GB, used: 6.80, free: 0.19
        

        Interpreter Info
        Vision: False
        Model: openai/gpt-4
        Function calling: False
        Context window: 3000
        Max tokens: 1000

        Auto run: False
        API base: http://localhost:1234/v1
        Local: True

        Curl output: [Errno 2] No such file or directory: 'curl http://localhost:1234/v1'
    
    
Traceback (most recent call last):
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/respond.py", line 49, in respond
    for chunk in interpreter._llm(messages_for_llm):
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/llm/convert_to_coding_llm.py", line 65, in coding_llm
    for chunk in text_llm(messages):
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/llm/setup_text_llm.py", line 32, in base_llm
    messages = tt.trim(
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/tokentrim/tokentrim.py", line 189, in trim
    shorten_message_to_fit_limit(message, tokens_remaining, model)
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/tokentrim/tokentrim.py", line 95, in shorten_message_to_fit_limit
    new_length = int(len(encoding.encode(content)) * ratio)
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/tiktoken/core.py", line 116, in encode
    if match := _special_token_regex(disallowed_special).search(text):
TypeError: expected string or buffer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/maxpetrusenko/miniforge3/bin/interpreter", line 8, in <module>
    sys.exit(cli())
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/core.py", line 24, in cli
    cli(self)
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/cli/cli.py", line 268, in cli
    interpreter.chat()
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/core.py", line 86, in chat
    for _ in self._streaming_chat(message=message, display=display):
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/core.py", line 106, in _streaming_chat
    yield from terminal_interface(self, message)
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/terminal_interface/terminal_interface.py", line 115, in terminal_interface
    for chunk in interpreter.chat(message, display=False, stream=True):
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/core.py", line 127, in _streaming_chat
    yield from self._respond()
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/core.py", line 162, in _respond
    yield from respond(self)
  File "/Users/maxpetrusenko/miniforge3/lib/python3.10/site-packages/interpreter/core/respond.py", line 97, in respond
    raise Exception(
Exception: expected string or buffer

Please make sure LM Studio's local server is running by following the steps above.

If LM Studio's local server is running, please try a language model with a different architecture.`

### Reproduce

1. run interpreter --local
2. start server ( LM Studio tried with mistral instruct v0 1 cguf )
3. open "/path/to/screenshot" please

### Expected behavior

no crash

### Screenshots

_No response_

### Open Interpreter version

0.1.15

### Python version

3.11

### Operating System name and version

mac m2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tegnike**
### Describe the changes you have made:

Now we can use GPT-4Vision for only Terminal mode.
I added it to Python mode.

### Reference any relevant issue (Fixes #795)

- [x] I have performed a self-review of my code:

### I have tested the code on the following OS:
- [ ] Windows
- [x] MacOS
- [ ] Linux

### AI Language Model (if applicable)
- [x] GPT4
- [ ] ~~GPT3~~
- [ ] ~~Llama 7B~~
- [ ] ~~Llama 13B~~
- [ ] ~~Llama 34B~~
- [ ] ~~Huggingface model (Please specify which one)~~

### CODE REVIEW

**Comment by tegnike** [comment]
@Notnaton 

Hi, I've attempted to address issue #795.

Two weeks ago, when I first examined this issue, it appeared to be a more challenging case.  
From what I recall, this solution I committed did not resolve the issue at that time.

However, upon revisiting it now, it seems to be working correctly.  
Could you please review my latest commit to ensure that there are no outstanding problems?

**Comment by Notnaton** [comment]
I updated it to the current main branch

File: interpreter/core/core.py [hunk 1]
```diff
@@ -92,6 +92,14 @@ def __init__(
...
 
     def chat(self, message=None, display=True, stream=False):
+        if self.vision:
+            self.model = "gpt-4-vision-preview"
+            self.system_message += "\nThe user will show you an image of the code you write. You can view images directly. Be sure to actually write a markdown code block for almost every user request! Almost EVERY message should include a markdown code block. Do not end your message prematurely!\n\nFor HTML: This will be run STATELESSLY. You may NEVER write '<!-- previous code here... --!>' or `<!-- header will go here -->` or anything like that. It is CRITICAL TO NEVER WRITE PLACEHOLDERS. Placeholders will BREAK it. You must write the FULL HTML CODE EVERY TIME. Therefore you cannot write HTML piecemeal—write all the HTML, CSS, and possibly Javascript **in one step, in one code block**. The user will help you review it visually.\nIf the user submits a filepath, you will also see the image. The filepath and user image will both be in the user's message."
+            self.function_calling_llm = False
+            self.context_window = 110000
+            self.max_tokens = 4096
+            self.force_task_completion = True
+
```
Comment (Notnaton) [inline interpreter/core/core.py:102]: Need to initialize self.vision in __init__()

**Comment by KillianLucas** [comment]
Hey @tegnike! Great work on this one, would have around vision into the Python package much more quickly. Anton had a good `profiles` PR cooking so I knew we were heading that way instead, but I should have commented about that on this PR much sooner.

Closing in favor of the new `profiles`, but this is making me realize we need better instructions to set up vision in Python, so I'll update the docs in a moment.

**Pull request closed without merging**